lucet-runtime/lucet-runtime-internals/src/region/mmap.rs

```diff
@@ -114,6 +106,17 @@ impl RegionInternal for MmapRegion {
         }
         module.validate_runtime_spec(&limits)?;
 
+        let slot = self
+            .freelist
+            .write()
+            .unwrap()
+            .pop()
+            .ok_or(Error::RegionFull(self.capacity))?;
+
+        if slot.heap as usize % host_page_size() != 0 {
+            lucet_bail!("heap is not page-aligned; this is a bug");
```

Comment:
Sorry I didn't catch this before, but this is also an early return. Since it would reflect a bug in Lucet, I think we should change it to a panic/assert rather than returning an `InternalError`.

Reply:
Thanks for catching this!

Reply:
Since the code you point at is in master, and since it feels a bit unrelated to my main task of customizing limits, I wonder if you might accept my fixing it in a PR over here:  https://github.com/bytecodealliance/lucet/pull/486.

Reply:
I left feedback there, but it is related to this PR because the early return at that location would cause the slot to leak.

Reply:
... sorry, I got confused about which PR I was looking at. Not the one we already merged :blush:

Reply:
No worries.  I'm having trouble keeping the little changes straight as well.